src/ssl.c

```diff
@@ -54583,16 +54583,25 @@ WOLFSSL_API PKCS7* wolfSSL_SMIME_read_PKCS7(WOLFSSL_BIO* in,
                 }
             }
 
-            canonPos--;
+            if (canonPos > 0) {
+                canonPos--;
+            }
+
             /* Strip the final trailing newline.  Support \r, \n or \r\n. */
             if (canonSection[canonPos] == '\n') {
-                canonPos--;
-                if (canonSection[canonPos] == '\r') {
+                if (canonPos > 0) {
                     canonPos--;
                 }
+                if (canonSection[canonPos] == '\r') {
```

Comment:
Don't need this as a separate case.
Remove this and instead of 'else if' replace with 'if'.

Reply:
check